### ISSUE TITLE
Dashboard search doesn't always match kwargs

### ISSUE THREAD

**Issue by gap777** [bensheldon/good_job#1547]
We sprinkle metadata into our job kwargs to enable lookup in the Dashboard, like this:

```
{"ad_id"=>"ad_qTB3kTjQeZEmptyx8JZ5", "ad_ref"=>"A-1012679", "line_item_gid"=>"gid://supply-side-platform/Orders::Types::ReservationLineItem/rli_Dd2LpagvXuqaZNAZt4yu", "purchase_order_id"=>"po_8o9yOwc88sfMQWFmjE8F", "_aj_ruby2_keywords"=>["purchase_order_id", "line_item_gid", "ad_ref", "ad_id"]}
```

* Searching for `po_8o9yOwc88sfMQWFmjE8F` finds the job.
* Searching for `8o9yOwc88sfMQWFmjE8F` (a substring of the previous arg) finds the job
* Searching for `A-1012679` does NOT find the job
* Searching for `gid://supply-side-platform/Orders::Types::ReservationLineItem/rli_Dd2LpagvXuqaZNAZt4yu` does NOT.
* Searching for `rli_Dd2LpagvXuqaZNAZt4yu` (a substring of the previous arg) does NOT.

Why can some args (`po_8o9yOwc88sfMQWFmjE8F`) be used for searching and others (`A-1012679`) can not be?

**Comment by bensheldon** [comment]
I think this might be a side effect of Postgres Full Text search. This is the query I'm using:

https://github.com/bensheldon/good_job/blob/b557525a9003a1fc1a434742dae58053b947e708/app/models/concerns/good_job/filterable.rb#L37-L42

Maybe `english` is the wrong stemmer to use, or something else.

**Comment by bensheldon** [comment]
yep! I think we should try it. I feel pretty good about the current regression tests, so we can be creative. I think probably the complexity of these arguments can be expanded to cover your needs:

https://github.com/bensheldon/good_job/blob/b557525a9003a1fc1a434742dae58053b947e708/spec/app/models/concerns/good_job/filterable_spec.rb#L8-L23

...and then we just gotta find a query that matches them.
